pyvista/plotting/render_window_interactor.py

```diff
@@ -433,26 +441,58 @@ def key_press_event(self, *args):
                 func()
 
     def update_style(self):
-        """Update the camera interactor style."""
-        if self._style_class is None:
-            # We need an actually custom style to handle button up events
-            self._style_class = _style_factory(self._style)(self)
-        self.interactor.SetInteractorStyle(self._style_class)
+        """Update the camera interactor style.
+
+        Called when setting interactor_style attribute.
+        """
+        self.interactor.SetInteractorStyle(self.style)
 
     @property
-    def style(self):
-        """Return the current interactor style.
+    def style(
+        self,
+    ) -> Optional[
+        Union[_vtk.vtkContextInteractorStyle, _vtk.vtkInteractorStyle, _CustomInteractorStyle]
+    ]:
+        """Get/set the current interactor style.
+
+        .. warning::
+
+            Setting an interactor style needs careful control of events handling.
```

Comment:
We could also consider exposing `_CustomInteractorStyle` as public and put in the docs explicitly.  Users could then subclass from it to get the recommended PyVista functionality for interactor styles.  Looking for feedback on this.

Reply:
This is now public and explicitly referenced in this warning.